### ISSUE TITLE
Query by ObjectId

### ISSUE THREAD

**Issue by patriciunista** [Microsoft/vscode-cosmosdb#644]
Repro steps:

The following query does not works:
`db.companies.find( { "_id": ObjectId('5b06c0aa64d4cb7723ce1229') } );`

it throws an error:
*Action: cosmosDB.executeMongoCommand*
*Error type: SyntaxError*
*Error Message: Unexpected token O in JSON at position 7*

Neither if I quote the function (as seen in other issues with `$set` parameter):

`db.companies.find( { "_id": "ObjectId('5b06c0aa64d4cb7723ce1229')" } );`

The response is an empty array. If I try with Robo 3T for example, it returns the following:
```json
{
    "_id" : ObjectId("5b06c0aa64d4cb7723ce1229"),
    "name" : "example"
}
```

Version: 0.7.1
OS: OS X 10.13.4

**Comment by PrashanthCorp** [comment]
Hi Patriciu, 

Thanks for your input! We currently don't parse ObjectID as-is. We can parse [EJSON](https://github.com/mongodb-js/extended-json). Your issue looks related to #564. Hope that gives you more information. 

If you're querying and ObjectID, you can try using a `$oid` tag like `db.companies.find( { "_id": "$oid": {'5b06c0aa64d4cb7723ce1229'} } );` . Let me know if that helps solve your problem.

**Comment by StephenWeatherford** [comment]
@PrashanthCorp Was there a way to use EJSON to parse ObjectId/et al, and possibly support both on input (but only the $oid version on output)?

**Comment by PrashanthCorp** [comment]
Closing as I feel it has been fixed via #635 

Feel free to reopen

**Comment by PrashanthCorp** [comment]
Hi @patriciunista , the merged fix was published today (version 0.8.0). Hope this fixes your issue. :-)
Let us know how the fix holds up with your use-case.
